apps/next/src/pages/Settings/components/NetworkManagement/hooks/useAddNetworks.ts

```diff
@@ -0,0 +1,76 @@
+import { NetworkVMType } from '@avalabs/core-chains-sdk';
+import { isNetworkValid } from '../components/NetworkForm/utils/isNetworkValid';
+import { AdvancedNetworkConfig, Network } from '@core/types';
+import { useCallback, useMemo, useState } from 'react';
+import { useAnalyticsContext, useNetworkContext } from '@core/ui';
+import { NetworkFormFieldInfo } from '../components/NetworkForm/types';
+import { DynamicFields, mergeDynamicFields } from './utils/fieldInfo';
+
+const defaultNetworkValues: Network & AdvancedNetworkConfig = {
+  chainName: '',
+  chainId: 0,
+  vmName: NetworkVMType.EVM,
+  rpcUrl: '',
+  networkToken: {
+    name: '',
+    symbol: '',
+    description: '',
+    decimals: 18,
+    logoUri: '',
+  },
+  logoUri: '',
+  explorerUrl: '',
+  customRpcHeaders: undefined,
+};
+
+export const useAddNetwork = () => {
+  const { saveCustomNetwork } = useNetworkContext();
+  const { capture } = useAnalyticsContext();
+
+  const [network, setNetwork] = useState<Network & AdvancedNetworkConfig>(
+    defaultNetworkValues,
+  );
+  const { isValid, errors } = useMemo(() => {
+    return isNetworkValid(network);
+  }, [network]);
```

Comment:
Something that Bartosz made me alert to -- `useMemo` is not needed for things like this.

Whenever you update any field, `network` will change and therefore `useMemo` will have to run again, giving us no benefit, but we'll still be paying the cost of memoization.